### ISSUE TITLE
Adding custom label should automatically switch label mode to "Custom"

### ISSUE THREAD

**Issue by matterhorn103** [OpenChemistry/avogadrolibs#1574]
When adding a custom text label using the Atom Label Tool, the typed text disappears after text entry is finished and only the element letter is shown. Clicking the atom again makes it reappear, but after pressing enter it always disappears.

The way to make the custom text visible is of course to firstly manually enable the "Label" display type, then go to the View Configuration pane and change the "Atom Labels" option (which should maybe be better called "Label Type")  to "Custom".

This is not however immediately obvious and so the apparent behaviour of adding custom labels having no effect is confusing.

I suggest simply that the Label display type is automatically enabled and switched to "Custom", if not already, when a custom label is added.

Alternatively this could be done whenever the Atom Label Tool is selected, but I suspect there are use cases where this might be undesirable.

**Comment by ghutchis** [comment]
At the moment, this is tricky, since the tool can't force a change to the rendering code.

What I *can* imagine is that the renderer checks for "Custom" and switch it on if labels are found. I need to think a bit about the best way to do this, likely with a new "molecule changed" flag for labels.

**Comment by matterhorn103** [comment]
I don't understand the code structure all that well, but does the tool need to change the rendering code? Can't it just emit a Signal that triggers the same Slot as is triggered when a different option is chosen from the View Configuration: Atom Labels dropdown?

**Comment by ghutchis** [comment]
No, because to trigger a slot, it needs to connect to the specific object. The tool has no good way to access that object.

But as I said, there's a functionally equivalent mechanism.

**Comment by ghutchis** [comment]
Is this still a bug? I think I fixed this along the way.

**Comment by matterhorn103** [comment]
Unless there has been some change since the 5th January, this is still the current behaviour (I checked with the Flatpak)
